File: CPP/PerfectNumber.cpp
#include <iostream>
#include <cmath>

using namespace std;

bool isPerfectNumber(int number){
    if(number <= 1){
        return false;
    }
    int sumOfDivisors = 1;
    for(int divisor = 2; divisor <= sqrt(number); divisor++){
        if(number % divisor == 0){
            sumOfDivisors += divisor;
            sumOfDivisors += number / divisor;
        }
        if(number == divisor * divisor){
            sumOfDivisors -= divisor;
        }
    }
    if (sumOfDivisors == number){
        return true;
    }
    return false;
}

int main() {
    cout << isPerfectNumber(6) << endl; //1+2+3=6
    cout << isPerfectNumber(28) << endl; //1+2+4+7+14=28
    cout << isPerfectNumber(12) << endl;
    cout << isPerfectNumber(25) << endl;
}

File: Python/PerfectNumber.py
#O(sqrt(n))
def is_perfect_number(number: int) -> bool:
    if number <= 1:
        return False
    sum_of_divisors = 1
    for divisor in range(2, int(number ** 0.5) + 1):
        if number % divisor == 0:
            sum_of_divisors += divisor
            sum_of_divisors += number // divisor
        if number == divisor * divisor:
            sum_of_divisors -= divisor
    if sum_of_divisors == number:
        return True
    return False

print(is_perfect_number(6)) # 1+2+3=6
print(is_perfect_number(28)) # 1+2+4+7+14=28
print(is_perfect_number(12))
print(is_perfect_number(25))